internal/migrations/migrator.go

```diff
@@ -737,3 +784,10 @@ func CleanupResolutionsAfterMigration(ctx context.Context, db sql.DB, adjustExpi
 
 	return tx.Commit(ctx)
 }
+
+type ConsensusParamsGetter func(ctx context.Context, height *int64) *cmtTypes.ConsensusParams
+
+func (m *Migrator) SetConsensusParamsGetter(fn ConsensusParamsGetter) {
+	m.consensusParamsFn = fn
+	m.consensusParamsFnChan <- true
```

Comment:
Just FYI, closing the channel would be another approach.  It really doesn't need to be anything besides an unbuffered `chan struct{}` to get the job done.

Also, without docs on this method it's quite mysterious why we're doing what we're doing here.  Maybe short docs here with "See also"s for the related code.